src/xcode/ENA/ENA/Source/Services/PPAnalyticsSubmitter/__tests__/PPAnalyticsSubmitterTests.swift

```diff
@@ -0,0 +1,320 @@
+
+		store.isPrivacyPreservingAnalyticsConsentGiven = true
+		store.lastSubmissionAnalytics = Calendar.current.date(byAdding: .day, value: -5, to: Date())
+		store.dateOfAcceptedPrivacyNotice = Calendar.current.date(byAdding: .day, value: -5, to: Date())
+		store.lastAppReset = Calendar.current.date(byAdding: .day, value: -5, to: Date())
+		let ppacToken = PPACToken(apiToken: "FakeApiToken", deviceToken: "FakeDeviceToken")
+
+		// WHEN
+		analyticsSubmitter.triggerSubmitData(ppacToken: ppacToken, completion: { result in
+			switch result {
+			case .success:
+				expectation.fulfill()
+			case .failure:
+				XCTFail("Test should not fail")
+			}
+		})
+
+		// THEN
+		waitForExpectations(timeout: .short)
+	}
+
+	// MARK: - Failures
+
+	func testGIVEN_SubmissionIsTriggered_WHEN_UserConsentIsMissing_THEN_UserConsentErrorIsReturned() {
+		// GIVEN
+		let store = MockTestStore()
+		let client = ClientMock()
+		let config = SAP_Internal_V2_ApplicationConfigurationIOS()
+		let appConfigurationProvider = CachedAppConfigurationMock(with: config)
+		let analyticsSubmitter = PPAnalyticsSubmitter(
+			store: store,
+			client: client,
+			appConfig: appConfigurationProvider
+		)
+
+		let expectation = self.expectation(description: "completion handler is called with an error")
+
+		// WHEN
+		var ppasError: PPASError?
+		analyticsSubmitter.triggerSubmitData(ppacToken: nil, completion: { result in
+			switch result {
+			case .success:
+				XCTFail("Test should not success")
+			case let .failure(error):
+				ppasError = error
+				expectation.fulfill()
+			}
+		})
+
+		// THEN
+		waitForExpectations(timeout: .short)
+		XCTAssertEqual(ppasError, .userConsentError)
+	}
+
+	func testGIVEN_SubmissionIsTriggered_WHEN_AppConfigIsMissing_THEN_ProbibilityErrorIsReturned() {
+		// GIVEN
+		let store = MockTestStore()
+		let client = ClientMock()
+		let config = SAP_Internal_V2_ApplicationConfigurationIOS()
+		let appConfigurationProvider = CachedAppConfigurationMock(with: config)
+		let analyticsSubmitter = PPAnalyticsSubmitter(
+			store: store,
+			client: client,
+			appConfig: appConfigurationProvider
+		)
+
+		let expectation = self.expectation(description: "completion handler is called with an error")
+		store.isPrivacyPreservingAnalyticsConsentGiven = true
+
+		// WHEN
+		var ppasError: PPASError?
+		analyticsSubmitter.triggerSubmitData(ppacToken: nil, completion: { result in
+			switch result {
+			case .success:
+				XCTFail("Test should not success")
+			case let .failure(error):
+				ppasError = error
+				expectation.fulfill()
+			}
+		})
+
+		// THEN
+		waitForExpectations(timeout: .short)
+		XCTAssertEqual(ppasError, .probibilityError)
+	}
+
+	func testGIVEN_SubmissionIsTriggered_WHEN_ProbabilityIsLow_THEN_ProbibilityErrorIsReturned() {
+		// GIVEN
+		let store = MockTestStore()
+		let client = ClientMock()
+		var config = SAP_Internal_V2_ApplicationConfigurationIOS()
+		config.privacyPreservingAnalyticsParameters.common.probabilityToSubmit = -1
```

Comment:
As above, why not in 0...1 or is this the idea of the test? The test's name is ambiguous 🤷‍♂️

Reply:
I wanted that the probability fails always, so the value has to be < 0. If not, I would produce a flaky test.